src/main/kotlin/app/revanced/patches/youtube/layout/startupshortsreset/patch/DisableShortsOnStartupPatch.kt

```diff
@@ -0,0 +1,72 @@
+package app.revanced.patches.youtube.layout.startupshortsreset.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.impl.BytecodeData
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.fingerprint.method.impl.MethodFingerprint.Companion.resolve
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.DependsOn
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.patch.impl.BytecodePatch
+import app.revanced.patches.music.layout.premium.fingerprints.HideGetPremiumFingerprint
+import app.revanced.patches.music.layout.premium.fingerprints.HideGetPremiumParentFingerprint
+import app.revanced.patches.youtube.layout.startupshortsreset.annotations.StartupShortsResetCompatibility
+import app.revanced.patches.youtube.layout.startupshortsreset.fingerprints.ActionOpenShortsFingerprint
+import app.revanced.patches.youtube.misc.integrations.patch.IntegrationsPatch
+import app.revanced.patches.youtube.misc.settings.bytecode.patch.SettingsPatch
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.StringResource
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.SwitchPreference
+import org.jf.dexlib2.iface.instruction.OneRegisterInstruction
+import org.jf.dexlib2.iface.instruction.TwoRegisterInstruction
+
+@Patch
+@DependsOn([IntegrationsPatch::class, SettingsPatch::class])
+@Name("disable-startup-shorts-player")
+@Description("Disable playing YouTube Shorts when launching YouTube.")
+@StartupShortsResetCompatibility
+@Version("0.0.1")
+class DisableShortsOnStartupPatch : BytecodePatch(
+    listOf(
+        ActionOpenShortsFingerprint
+    )
+) {
+    override fun execute(data: BytecodeData): PatchResult {
+        SettingsPatch.PreferenceScreen.LAYOUT.addPreferences(
+            SwitchPreference(
+                "revanced_startup_shorts_player_enabled",
+                StringResource("revanced_startup_shorts_player_title", "Disable shorts player at app startup"),
+                false,
+                StringResource("revanced_startup_shorts_player_summary_on", "Shorts player is disabled at app startup"),
+                StringResource("revanced_startup_shorts_player_summary_off", "Shorts player is enabled at app startup")
+            )
+        )
+
+        val actionOpenShortsResult = ActionOpenShortsFingerprint.result
+        val actionOpenShortsMethod = actionOpenShortsResult!!.mutableMethod
+        val actionOpenShortsIstructions = actionOpenShortsMethod.implementation!!.instructions
+
+        val moveResultIndex = actionOpenShortsResult.scanResult.patternScanResult!!.endIndex + 21
```

Comment:
> 21 is an arbitrary offset, capture the index with an opcode pattern.

Too many opcodes to capture. Fingerprint doesn't work.

Reply:
Then capture the opcodes from that region instead of where you do now.

Reply:
> Then capture the opcodes from that region instead of where you do now.

I can just take the index with an indexOfFirst on the const-string opcode.

Just that...

Reply:
You should use a pattern which includes the instruction your offset points to.

Reply:
> You should use a pattern which includes the instruction your offset points to.

I need of examples, otherwise I don't know of what you're talking about. I got tired of trying to understand, you have to be clearer.

Reply:
Your current pattern is arbitrary and just serves the purpose of finding the method. Use a pattern which scanResult is also useful, theres plenty examplary patches you can look into where the scanResult is used.

Reply:
> Your current pattern is arbitrary and just serves the purpose of finding the method. Use a pattern which scanResult is also useful, theres plenty examplary patches you can look into where the scanResult is used.

ScanResult give me index 0, it's totally useless.

Reply:
Open an issue.